git_machete/client.py

```diff
@@ -747,6 +747,8 @@ def traverse(
                         self.__down_branches[upstream])
                     if branch in self.__annotations:
                         del self.__annotations[branch]
+                    if branch in self.managed_branches:
```

Comment:
Could you provide a regression test that would've captured the bug?

Reply:
Oh! okay... could you remind me how you discovered the bug in the first place?

Reply:
I think i was debugging the `traverse` function and saw that `self.managed_branches` wasn't updated

Reply:
Ok, let's merge as is then